config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/PolicyType.java

```diff
@@ -15,27 +17,37 @@
 import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 public enum PolicyType {
-    INCLUDE("Include"),
-    SERVICE_OPERATION("Service Operation"),
-    GLOBAL("Global"),
-    INTERNAL("Internal",
+    INCLUDE("Include", Policy.class),
+    SERVICE_OPERATION("Service Operation", Policy.class),
+    GLOBAL("Global", GlobalPolicy.class),
+    INTERNAL("Internal", AuditPolicy.class,
             "audit-lookup",
             "audit-sink",
             "audit-message-filter",
             "audit-viewer");
 
     private String type;
     private List<String> tags;
+    private Class<? extends Policy> policyClass;
 
-    PolicyType(String type, String... tags) {
+    PolicyType(String type, Class<? extends Policy> policyClass, String... tags) {
         this.type = type;
+        this.policyClass = policyClass;
         this.tags = new ArrayList<>(asList(firstNonNull(tags, new String[0])));
     }
 
     public String getType() {
         return type;
     }
 
+    public Policy createPolicyObject() {
+        try {
+            return policyClass.newInstance();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "newInstance"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=125&issues=AWbwkFSATJjFgJLTCSN5&open=AWbwkFSATJjFgJLTCSN5)

[ISSUE_KEY]: AWbwkFSATJjFgJLTCSN5